packages/grz-common/src/grz_common/transfer.py

```diff
@@ -76,3 +78,26 @@ def init_s3_resource(s3_options: S3Options) -> S3ServiceResource:
     )
 
     return s3_resource
+
+
+def get_version_info(s3_options: S3Options, version_file_path: str) -> str | None:
+    """Download the version file from S3 and return its contents."""
+    try:
+        s3_client = init_s3_client(s3_options)
+
+        # download the version file content directly to memory
+        response = s3_client.get_object(Bucket=s3_options.bucket, Key=version_file_path)
+        version_content = response["Body"].read().decode("utf-8").strip()
+        version_data = json.loads(version_content)
+
+        # extract values
+        minimal_version = version.parse(version_data["minimal_version"])
+        latest_version = version.parse(version_data["latest_version"])
+        return minimal_version, latest_version
```

Comment:
Perhaps we can define and use a little pydantic model class for this VersionFile instead, should be a little bit more robust and easier to find/read.